### DISCUSSION TITLE
R.nvim linting support

### DISCUSSION THREAD

**Discussion by ssh352** [R-nvim/R.nvim#134] [Q&A] [answered]
with Nvim-r plugin I used https://github.com/neoclide/coc-r-lsp for lint errors:
```
r.lsp.diagnostics: Enable linting of R code, using the lintr package. Defaults to true. To disable this, you must have at least version 0.2.7 of the R Language Server installed.
```

Now I almost migrated to LazyVim and R.nvim. Does R.nvim supports linting diagnostics through lintr package? Thanks!

**Comment by jalvesaq** [comment, marked as answer]
No, it doesn't. I don't know about other `R.nvim` developers, but I don't plan to implement linting diagnostics because it's a feature already available through the `languageserver`.

**Reply by ssh352** [reply]
do I still have to use  https://github.com/neoclide/coc-r-lsp for linting? Do you have suggestion what to use? I thought after moving to R.nvim from nvim-R I can get rid of coc.

**Comment by jalvesaq** [comment]
You can try [nvim-lspconfig](https://github.com/neovim/nvim-lspconfig). The internet has many examples of how to use it.

I posted my complete Neovim setup some time ago on https://github.com/R-nvim/cmp-r/issues/5#issuecomment-2000450625 including `~/.config/nvim/lua/plugins/lsp.lua`.

**Reply by ssh352** [reply]
verified, it works! Actually on the fly linting works even without R.nvim. Didn't know that. Thanks!

**Comment by PMassicotte** [comment]
Better to use `lintr` directly.

**Reply by ssh352** [reply]
Yes I use both on the fly linting and lintr::lint_package() for final lint check.
